clippy_lints/src/matches.rs

```diff
@@ -850,6 +989,7 @@ where
     }
 
     impl<'a, T: Copy> Kind<'a, T> {
+        #[allow(clippy::match_single_binding)]
```

Comment:
This shouldn't be required that often. Looking at [`PatKind`](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_hir/enum.PatKind.html) we should be able to easily figure out these FPs:

Only irrefutable patterns should get linted. Luckily we already have a function for this:
https://github.com/rust-lang/rust-clippy/blob/e36a33fac779458b65ae6b45c692a1700b08e506/clippy_lints/src/utils/mod.rs#L849-L850

Please write tests for the more complex cases, like `Struct`/`Or`/`Tuple`, ... and make sure that the suggestions are correct.

If you don't want to put that much more work into this lint, you can limit this lint to `PatKind::Binding` and `PatKind::Tuple` for now, which should be the easiest cases.

Reply:
In any case, make sure to not trigger this lint for `Or` patterns, since these are feature gated and this feature is incomplete.

Reply:
Last note: `PatKind::Wild` could be sepcial cased, so that no `let _ = ..` is suggested, but only the match arm body.